components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1404,14 +1417,14 @@ void KeyringService::AddHardwareAccounts(
     }
 
     meta_value->SetKey(info->address, std::move(hw_account));
-
+    addresses.push_back(info->address);
     if (!account_selected) {
       SetSelectedAccountForCoinSilently(infos[0]->coin, infos[0]->address);
       SetSelectedCoin(prefs_, infos[0]->coin);
       account_selected = true;
     }
   }
-
+  json_rpc_service_->DiscoverAssets(mojom::kMainnetChainId, addresses);
```

Comment:
In that function, please document the trigger points we currently have.

Reply:
Also, seems like we're missing ImportAccount?

Reply:
I left it out because ImportAccount calls ImportAccountForKeyring which makes the DiscoverAssets call https://github.com/brave/brave-core/blob/4392ec2657ee1dd32fb993f0ef0b8ca310efd1c1/components/brave_wallet/browser/keyring_service.cc#L1019